python/cuspatial/cuspatial/geometry/geocolumn.py

```diff
@@ -0,0 +1,491 @@
+        object.
+        """
+        return self._meta.input_types
+
+    @_types.setter
+    def _types(self, types):
+        raise TypeError("GeoPandasMeta does not support item assignment.")
+
+    @property
+    def _lengths(self):
+        """
+        A list of integers of the length of each Multi geometry. Each non-multi
+        geometry is length 1.
+        """
+        return self._meta.input_lengths
+
+    @_lengths.setter
+    def _lengths(self, lengths):
+        raise TypeError("GeoPandasMeta does not support item assignment.")
+
+    def __len__(self):
+        """
+        Returns the number of unique geometries stored in this GeoColumn.
+        """
+        return len(self._meta.input_types)
+
+    @property
+    def points(self):
+        """
+        The Points column is a simple numeric column. x and y coordinates
+        can be stored either interleaved or in separate columns. If a z
+        coordinate is present, it will be stored in a separate column.
+        """
+        return self._geo._points
+
+    @property
+    def multipoints(self):
+        """
+        The MultiPoints column is similar to the Points column with the
+        addition of an offsets column. The offsets column stores the comparable
+        sizes and coordinates of each MultiPoint in the cuGeoSeries.
+        """
+        return self._geo._multipoints
+
+    @property
+    def lines(self):
+        """
+        LineStrings contain the coordinates column, an offsets column, and a
+        multioffsets column. The multioffsets column stores the indices of the
+        offsets that indicate the beginning and end of each MultiLineString
+        segment.
+        """
+        return self._geo._lines
+
+    @property
+    def polygons(self):
+        """
+        Polygons contain the coordinates column, a rings olumn specifying
+        the beginning and end of every polygon, a polygons column specifying
+        the beginning, or exterior, ring of each polygon and the end ring.
+        All rings after the first ring are interior rings.  Finally a
+        multipolys column stores the offsets of the polygons that should be
+        grouped into MultiPolygons.
+        """
+        return self._geo._polygons
+
+    def _dump(self):
+        return (
+            f"POINTS\n"
+            f"{self.points._repr__()}\n"
+            f"MULTIPOINTS\n"
+            f"{self.multipoints._repr__()}\n"
+            f"LINES\n"
+            f"{self.lines._repr__()}\n"
+            f"POLYGONS\n"
+            f"{self.polygons._repr__()}\n"
+        )
+
+    def copy(self, deep=True):
+        """
+        Create a copy of all of the GPU-backed data structures in this
+        GeoColumn.
+        """
+        result = GeoColumn(self._geo.copy(), self._meta.copy(), self.data)
+        return result
+
+    def _copy_type_metadata(self: T, other: ColumnBase) -> ColumnBase:
+        self._data = other.data
+        return self
+
+
+class gpuGeometry:
```

Comment:
Class names should typically begin with an uppercase letter with a [few exceptions](https://www.python.org/dev/peps/pep-0008/#class-names). Any reason to deviate?

Reply:
No